src/hepattn/models/maskformer.py

```diff
@@ -73,6 +77,10 @@ def __init__(
         self.raw_variables = raw_variables or []
         self.log_attn_mask = log_attn_mask
         self.log_step = 0
+        self.query_posenc = query_posenc
+        self.preserve_query_posenc = preserve_query_posenc
+        self.preserve_key_embed = preserve_key_embed
+        self.preserve_key_posenc = preserve_key_posenc
```

Comment:
Following from the other PR, do you think we can avoid `preserve_key_embed`? Also seems reasonable to harmonise `preserve_query_posenc` and `preserve_key_posenc`, what do you think?